bool        is_ball_detected
int32       discrete_turns_needed # 0 if right on the ball, negative value - turn left, positive - turn right.
uint32      detected_x
uint32      detected_y
uint32      detected_r
